x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotRecoveryStateIntegrationTests.java

```diff
@@ -98,22 +160,49 @@ public void testRecoveryStateRecoveredBytesMatchPhysicalCacheState() throws Exce
         assertExecutorIsIdle(SearchableSnapshotsConstants.CACHE_PREWARMING_THREAD_POOL_NAME);
         assertExecutorIsIdle(SearchableSnapshotsConstants.CACHE_FETCH_ASYNC_THREAD_POOL_NAME);
 
-        final RecoveryResponse recoveryResponse = client().admin().indices().prepareRecoveries(restoredIndexName).get();
-        Map<String, List<RecoveryState>> shardRecoveries = recoveryResponse.shardRecoveryStates();
-        assertThat(shardRecoveries.containsKey(restoredIndexName), equalTo(true));
-        List<RecoveryState> recoveryStates = shardRecoveries.get(restoredIndexName);
-        assertThat(recoveryStates.size(), equalTo(1));
-        RecoveryState recoveryState = recoveryStates.get(0);
+        RecoveryState recoveryState = getRecoveryState(restoredIndexName);
 
         assertThat(recoveryState.getStage(), equalTo(RecoveryState.Stage.DONE));
 
         long recoveredBytes = recoveryState.getIndex().recoveredBytes();
         long physicalCacheSize = getPhysicalCacheSize(restoredIndex, snapshotInfo.snapshotId().getUUID());
 
-        assertThat("Physical cache size doesn't match with recovery state data", physicalCacheSize, equalTo(recoveredBytes));
+        assertThat("Expected to reuse all data from the persistent cache but it didn't", 0L, equalTo(recoveredBytes));
+
+        final Repository repository = internalCluster().getDataNodeInstance(RepositoriesService.class).repository(fsRepoName);
+        assertThat(repository, instanceOf(BlobStoreRepository.class));
+        final BlobStoreRepository blobStoreRepository = (BlobStoreRepository) repository;
+
+        final RepositoryData repositoryData = ESBlobStoreRepositoryIntegTestCase.getRepositoryData(repository);
+        final IndexId indexId = repositoryData.resolveIndexId(indexName);
+        long inMemoryCacheSize = 0;
+        for (int shardId = 0; shardId < numberOfShards; shardId++) {
+            final BlobStoreIndexShardSnapshot snapshot = blobStoreRepository.loadShardSnapshot(
+                blobStoreRepository.shardContainer(indexId, shardId),
+                snapshotInfo.snapshotId()
+            );
+            inMemoryCacheSize += snapshot.indexFiles()
```

Comment:
I supposed that we could compute the `physicalCacheSize` from the BlobStoreIndexShardSnapshot too? Maybe also check that `getPhysicalCacheSize()` is equal to `physicalCacheSize`?

Reply:
Addressed in https://github.com/elastic/elasticsearch/pull/67425/commits/e0c5c8d9eb3b271d669bdedc239ca255b0d510b6